packages/recipe/src/components/EzFlashMessage/EzFlashMessage.tsx

```diff
@@ -82,6 +83,24 @@ const icons = {
   success: <SuccessIcon />,
 };
 
+/**
+ * Gets the appropriate role given the props passed in
+ * 1. if this is an info dialog, we probably don't need to aggressively alert them. For this, we use `role="status"`
+ * 2. if it is dismissable, we can use `role="alertdialog"`. "If the user is expected to close the alert, then the alertdialog role should be used instead."
+ * 3. in all other cases, assume a default of `role="alert"`
+ * @see: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/alert_role
+ * @returns an accessible role for the DOM element containing the alert.
+ */
+function getRole({use, onDismiss}: Pick<FlashMessageProps, 'use' | 'onDismiss'>) {
+  if (use === 'info') {
+    return 'status' as const;
+  }
+  if (typeof onDismiss !== 'undefined') {
+    return 'alertdialog' as const;
+  }
+  return 'alert' as const;
+}
```

Comment:
This looks good (with the change @pgrippi suggested), however I'm not sure we should use `alertdialog` as a role here, where [initial focus goes to an element within the dialog](https://www.digitala11y.com/alertdialog-role/). Since we aren't using a modal/dialog here, I don't think this applies. It should probably just be `alert` if not `info` use. What do you think?

```suggestion
const getRole = ({use}: Pick<FlashMessageProps, 'use'>): AriaRole =>
  use === 'info' ? 'status' : 'alert';
```

You might just inline this now that it's this small.

Reply:
Wouldn't we have to nest a ternary for that?